### DISCUSSION TITLE
DynamicFilterInfo支持JArray,JObject字段嘛

### DISCUSSION THREAD

**Discussion by netssfy** [dotnetcore/FreeSql#1819] [Q&A]
如题，DynamicFilterInfo支持JArray,JObject字段嘛
比如我有个JObject，字段名叫Attrs，他的内容是{"attr a":"value a"}
如果支持的话应该怎么构造DynamicFilterInfo，去实现Attrs["attr a"] == "value a"这样的条件

**Comment by 2881099** [comment]
自定义 Custom 解决一下。

**Reply by netssfy** [reply]
能贴一下SAMPLE代码吗，我试试。我对整个框架还不太熟悉，不太清楚CUSTOM的入口在哪里

**Reply by 2881099** [reply]
https://github.com/dotnetcore/FreeSql/discussions/1550

**Reply by netssfy** [reply]
完全照1550写了个SAMPLE
"FreeSql: Custom could not find the corresponding {re
flection information}:DynamicFilterCustomImpl.CustomLinq }}"

是要注册这个DynamicLinq函数吗？

**Reply by 2881099** [reply]
后面还有一个实现类，也要copy

**Reply by netssfy** [reply]
```
public class DynamicFilterCustomImpl
{
    //JSON Field 对应这个值
    public static string CustomLinq = $"{nameof(DynamicFilterCustomImpl.DynamicLinq)} {typeof(DynamicFilterCustomImpl).FullName},{typeof(DynamicFilterCustomImpl).Assembly.FullName}";

    [DynamicFilterCustom]
    public static LambdaExpression DynamicLinq(object sender, string value)
    {
        if (string.IsNullOrWhiteSpace(value)) value = "1==2";
        ParameterExpression t = Expression.Parameter(sender.GetType().GetGenericArguments()[0], "t");
        var exp = DynamicExpressionParser.ParseLambda(new ParameterExpression[] { t }, typeof(bool), value);
        return exp;
    }
}
```
这个嘛？我也COPY了. 但是看报错反射没能找到。Freesql是会自己在assembly里去找DynamicFilterCustom?

**Reply by netssfy** [reply]
![image](https://github.com/dotnetcore/FreeSql/assets/6095138/b0881c59-f9f6-47b7-a3b8-878b0a209961)

public static string CustomLinq = $"{nameof(DynamicFilterCustomImpl.DynamicLinq)} {typeof(DynamicFilterCustomImpl).FullName},{typeof(DynamicFilterCustomImpl).Assembly.FullName}";

这个CustomLinq的字符串是一种约定名嘛？还是说这里写错了？Field里填的是DynamicFilterCustomImpl.CustomLinq，是去找这个静态变量？然后通过这个静态变量去找这个DynamicLinq函数？

**Reply by 2881099** [reply]
CustomLinq 是一个反射信息，可以直接写死在 Field。
